### ISSUE TITLE
nested parameters in button_to incorrectly generate hidden fields

### ISSUE THREAD

**Issue by lifeiscontent** [rails/rails#25256]
### Steps to reproduce

in rails I have a `Project` model with an enum of stages using the following pattern in ERB.

``` erb
<% Project.stages.each do |key, value| %>
  <%= button_to key, project_path(@project), params: { project: { stage: value } }, method: :patch %>
<% end %>
```

### Expected behavior
the HTML should generate to this:

``` html
<form class="button_to" method="post" action="/projects/1">
  <input type="hidden" name="_method" value="patch">
  <input type="submit" value="New Prospect">
  <input type="hidden" name="authenticity_token" value="lSw51tROABF0KqhZMvYrbD7I6b0EepsTzzMEHRFMb66YV3Wz/mx/JeG0G2ovGRQjn+O8QcdckPctPX6Rn1w3QQ==">
  <input type="hidden" name="project[stage]" value="0">
</form>
```

### Actual behavior

the HTML generates to this:

``` html
<form class="button_to" method="post" action="/projects/1">
  <input type="hidden" name="_method" value="patch">
  <input type="submit" value="New Prospect">
  <input type="hidden" name="authenticity_token" value="lSw51tROABF0KqhZMvYrbD7I6b0EepsTzzMEHRFMb66YV3Wz/mx/JeG0G2ovGRQjn+O8QcdckPctPX6Rn1w3QQ==">
  <input type="hidden" name="project" value="stage=0">
</form>
```

### System configuration
**Rails version**: 4.2.5.1

**Ruby version**: ruby 2.2.4p230

**Comment by sgrif** [comment]
/cc @rafaelfranca 

This was fixed in 5.0 by #17043, unsure if we want to backport to 4.2 or not, since it's a minor breaking change

**Comment by rafaelfranca** [comment]
Yeah, we are not going to backport it. Thank you for reporting. Closing since it is fixed on master
